src/v/kafka/server/tests/alter_config_test.cc

```diff
@@ -886,3 +885,171 @@ FIXTURE_TEST(test_alter_config_internal_topic, alter_config_test_fixture) {
       incr_resp.data.responses[0].error_code,
       kafka::error_code::invalid_config);
 }
+
+FIXTURE_TEST(test_shadow_indexing_alter_configs, alter_config_test_fixture) {
+    wait_for_controller_leadership().get();
+    model::topic test_tp{"topic-1"};
+    create_topic(test_tp, 6);
+    using map_t = absl::flat_hash_map<ss::sstring, ss::sstring>;
+    std::vector<map_t> test_cases;
+
+    {
+        map_t properties;
+        properties.emplace("redpanda.remote.write", "false");
+        properties.emplace("redpanda.remote.read", "true");
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace("redpanda.remote.write", "true");
+        properties.emplace("redpanda.remote.read", "false");
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace("redpanda.remote.write", "true");
+        properties.emplace("redpanda.remote.read", "true");
+        test_cases.push_back(std::move(properties));
+    }
+    {
+        map_t properties;
+        properties.emplace("redpanda.remote.write", "false");
+        properties.emplace("redpanda.remote.read", "false");
+        test_cases.push_back(std::move(properties));
+    }
```

Comment:
I think it would be worth adding some test cases here that are noop for one of the configs to verify that they are not changed. Both for `alter_configs` and for `incremental_alter_configs` below.